src/wms/src/test/java/org/geoserver/wms/WMSVectorDimensionValidationTest.java

```diff
@@ -0,0 +1,443 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms;
+
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static org.geoserver.data.test.MockData.SF_PREFIX;
+import static org.geoserver.data.test.MockData.SF_URI;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.xml.namespace.QName;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.DimensionInfo;
+import org.geoserver.catalog.DimensionPresentation;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.impl.DimensionInfoImpl;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.ServiceException;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.locationtech.jts.geom.Envelope;
+
+/**
+ * this suite performs tests on the WMS request validation feature added for <a
+ * href="https://osgeo-org.atlassian.net/browse/GEOS-9757">GEOS-9757 - Return a service exception
+ * when client provided WMS dimensions are not a match</a>.
+ *
+ * @author skalesse
+ * @since 2024-07-11
+ */
+public class WMSVectorDimensionValidationTest extends WMSTestSupport {
+
+    private static final QName TIME_WITH_START_END =
+            new QName(SF_URI, "TimeWithStartEnd", SF_PREFIX);
+    private static final QName ELEVATION_WITH_START_END =
+            new QName(SF_URI, "ElevationWithStartEnd", SF_PREFIX);
+    private static final QName CUSTOMS_DIM_WITH_START_END =
+            new QName(SF_URI, "CustomDimensionWithStartEnd", SF_PREFIX);
+
+    private WMS wms;
+
+    @Before
+    public void setWMS() {
+        wms = getWMS();
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        String testDataFileName = "TimeElevationWithStartEnd.properties";
+        testData.addVectorLayer(
+                TIME_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+
+        testData.addVectorLayer(
+                ELEVATION_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+
+        testData.addVectorLayer(
+                CUSTOMS_DIM_WITH_START_END, emptyMap(), testDataFileName, getClass(), getCatalog());
+    }
+
+    /**
+     * for a 'time' dimension, if a matching value is provided in the request the validation should
+     * succeed
+     *
+     * @throws IOException an unexpected IO exception is thrown
+     */
+    @Test
+    public void matching_time_dimension_should_validate() throws IOException {
```

Comment:
We use camel case for method names (snake case is common in other programming languages). Please fix them all.

Reply:
Done.